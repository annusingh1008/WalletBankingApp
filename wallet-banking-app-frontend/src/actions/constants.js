export const authConstants = {
    LOGIN_REQUEST: 'LOGIN_REQUEST',
    LOGIN_SUCCESS: 'LOGIN_SUCCESS',
    LOGIN_FAILURE: 'LOGIN_FAILURE',
    LOGOUT_REQUEST: 'LOGOUT_REQUEST',
    LOGOUT_SUCCESS: 'LOGOUT_SUCCESS',
    LOGOUT_FAILURE: 'LOGOUT_FAILURE'
}

export const userConstants = {
    USER_REGISTER_REQUEST: 'USER_REGISTER_REQUEST',
    USER_REGISTER_SUCCESS: 'USER_REGISTER_SUCCESS',
    USER_REGISTER_FAILURE: 'USER_REGISTER_FAILURE',
}

export const amountTransferConstants = {
    AMOUNT_TRANSFER_REQUEST: 'AMOUNT_TRANSFER_REQUEST',
    AMOUNT_TRANSFER_SUCCESS: 'AMOUNT_TRANSFER_SUCCESS',
    AMOUNT_TRANSFER_FAILURE: 'AMOUNT_TRANSFER_FAILURE'
}

export const rechargeConstants = {
    RECHARGE_REQUEST: 'RECHARGE_REQUEST',
    RECHARGE_SUCCESS: 'RECHARGE_SUCCESS',
    RECHARGE_FAILURE: 'RECHARGE_FAILURE',
}

export const getUserDetailsConstants = {
    GET_USER_DETAILS_REQUEST: 'GET_USER_DETAILS_REQUEST',
    GET_USER_DETAILS_SUCCESS: 'GET_USER_DETAILS_SUCCESS',
    GET_USER_DETAILS_FAILURE: 'GET_USER_DETAILS_FAILURE'
}

export const transactionsConstants = {
    TRANSACTIONS_REQUEST: 'TRANSACTIONS_REQUEST',
    TRANSACTIONS_SUCCESS: 'TRANSACTIONS_SUCCESS',
    TRANSACTIONS_FAILURE: 'TRANSACTIONS_FAILURE',
    TOTAL_TRANSACTIONS_SUCCESS: 'TOTAL_TRANSACTIONS_SUCCESS',
}

export const cashbackConstants = {
    CASHBACK_REQUEST: 'CASHBACK_REQUEST',
    CASHBACK_SUCCESS: 'CASHBACK_SUCCESS',
    CASHBACK_FAILURE: 'CASHBACK_FAILURE',
    TOTAL_CASHBACKS_SUCCESS: 'TOTAL_CASHBACKS_SUCCESS'
}

export const getBalanceConstants = {
    GET_BALANCE_SUCCESS: 'GET_BALANCE_SUCCESS'
}