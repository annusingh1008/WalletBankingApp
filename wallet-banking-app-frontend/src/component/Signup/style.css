.label{
    margin-top: 20px;
}

.btn{
    margin-top: 25px;
}

.input{
    width: 300px;
}