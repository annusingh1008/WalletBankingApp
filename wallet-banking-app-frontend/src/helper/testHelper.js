import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";

const localStorageData = {
  email: "annu",
  password: "1234",
};

export const localStorageMock = (function () {
  return {
    getItem: function (key) {
      return localStorageData[key];
    },
    setItem: jest.fn(),
  };
})();

const mockStore = configureMockStore();

export const renderWithRedux = (ui, { store } = {}) => {
  return {
    ...render(<Provider store={mockStore(store)}>{ui}</Provider>),
    store,
  };
};
