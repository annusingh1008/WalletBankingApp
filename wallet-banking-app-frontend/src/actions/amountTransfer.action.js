import axios from "axios";
import { amountTransferConstants } from "./constants";

export const amountTransfer = (amountDetails) => {
  return async (dispatch) => {
    const res = await axios.post("http://localhost:8085/transfer/amount", {
      ...amountDetails,
    });

    if (res.status === 200) {
      if (res.data === "User does not exist") {
        alert("User does not exist! \nPlease check your email.");
      } else if (res.data === "Amount Transferred Successfully...!!") {
        alert("Transaction Successful");

        dispatch({
          type: amountTransferConstants.AMOUNT_TRANSFER_SUCCESS,
          payload: {
            amount:
              parseInt(amountDetails.currentAmount) -
              parseInt(amountDetails.amount),
            email: amountDetails.email,
          },
        });
      }
    }
  };
};
