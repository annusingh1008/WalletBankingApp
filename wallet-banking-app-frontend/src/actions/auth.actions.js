import axios from "axios";
import { authConstants } from "./constants";

export const login = (user) => {
  return async (dispatch) => {
    const res = await axios.post("http://localhost:8085/signin", {
      ...user,
    });

    if (res.status === 200) {
      if (res.data === "Signed in Successfully...!!") {
        localStorage.setItem("user", JSON.stringify(user));
        localStorage.setItem("email", user.email);

        dispatch({
          type: authConstants.LOGIN_SUCCESS,
          payload: user,
        });
      } else if (res.data === "Account does not exists with this email...!!") {
        alert("Account does not exists!! \nPlease check your email..");
      } else if (res.data === "Incorrect Password...!!") {
        alert("Please check your Password!!");
      }
    }
  };
};

export const getAuthDetails = (state) => state.auth;

export const signout = () => {
  return async (dispatch) => {
    dispatch({ type: authConstants.LOGOUT_REQUEST });
    localStorage.clear();
    dispatch({ type: authConstants.LOGOUT_SUCCESS });
  };
};
