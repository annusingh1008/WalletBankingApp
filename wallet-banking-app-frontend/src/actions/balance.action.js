import axios from "axios";
import { getBalanceConstants } from "./constants";

export const getUserBalance = (email) => {
  return async (dispatch) => {
    const res = await axios.get(`http://localhost:8085/getBalance/${email}`);

    dispatch({
      type: getBalanceConstants.GET_BALANCE_SUCCESS,
      payload: res.data,
    });
  };
};
