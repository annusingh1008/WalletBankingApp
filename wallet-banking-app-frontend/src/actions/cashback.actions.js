import axios from "axios";
import { cashbackConstants } from "./constants";

export const getAllCashBacks = (currentPage, email) => {
  return async (dispatch) => {
    const userEmail = localStorage.getItem("email");
    const res = await axios.get(
      `http://localhost:8085/getAllCashbacks/${userEmail}?pageNumber=${currentPage}&pageSize=${10}`
    );

    dispatch({
      type: cashbackConstants.CASHBACK_SUCCESS,
      payload: res.data,
    });
  };
};

export const getCashbacks = (state) => state.cashbacks;

export const getTotalCashbacks = (email) => {
  return async (dispatch) => {
    const res = await axios.get(
      `http://localhost:8085/getTotalCashbacks/${email}`
    );

    dispatch({
      type: cashbackConstants.TOTAL_CASHBACKS_SUCCESS,
      payload: res.data,
    });
  };
};
