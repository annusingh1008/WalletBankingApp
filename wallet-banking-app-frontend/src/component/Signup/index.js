import React, { useState, useEffect } from "react";
import { Container, Row, Col, Form, Button } from "react-bootstrap";
import { useDispatch, useSelector } from "react-redux";
import { getUser, signup } from "../../actions/user.actions";
import Layout from "../Layout";
import "./style.css";

const Signup = () => {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [firstName, setFirstName] = useState("");
  const [lastName, setLastName] = useState("");
  const [confirmPassword, setConfirmPassword] = useState("");
  const [mobileNumber, setMobileNumber] = useState("");

  const dispatch = useDispatch();
  const user = useSelector(getUser);

  // useEffect(() => {
  //   setFirstName("");
  //   setLastName("");
  //   setEmail("");
  //   setPassword("");
  //   setConfirmPassword("");
  //   setMobileNumber("");
  // }, [user.loading]);

  useEffect(() => {
    return () => {
      setFirstName("");
      setLastName("");
      setEmail("");
      setPassword("");
      setConfirmPassword("");
      setMobileNumber("");
    };
  }, [user.loading]);

  const userSignup = (e) => {
    e.preventDefault();
    if (password !== confirmPassword) {
      alert("Password does not match");
    } else {
      const user = {
        firstName,
        lastName,
        email,
        password,
        confirmPassword,
        mobileNumber,
      };

      dispatch(signup(user));
    }
  };

  return (
    <Layout>
      <Container style={{ marginTop: "110px" }}>
        <Row style={{ marginLeft: "18%" }}>
          <Col>
            <Form data-testid="submit-form" onSubmit={userSignup}>
              <label>First Name</label>
              <br />
              <input
                data-testid="inputFirstName"
                className="input"
                placeholder="First Name"
                type="text"
                value={firstName}
                onChange={(e) => setFirstName(e.target.value)}
                required
              />
              <br />
              <label className="label">Last Name</label>
              <br />
              <input
                data-testid="inputLastName"
                className="input"
                placeholder="Last Name"
                type="text"
                value={lastName}
                onChange={(e) => setLastName(e.target.value)}
                required
              />{" "}
              <br />
              <label className="label">Email</label>
              <br />
              <input
                data-testid="inputEmail"
                className="input"
                placeholder="Email"
                value={email}
                type="email"
                onChange={(e) => setEmail(e.target.value)}
                required
              />{" "}
              <br />
              <label className="label">Password</label>
              <br />
              <input
                data-testid="inputPassword"
                className="input"
                placeholder="Password"
                value={password}
                minLength="8"
                type="password"
                onChange={(e) => setPassword(e.target.value)}
                required
              />{" "}
              <br />
              <label className="label">Confirm Password</label>
              <br />
              <input
                data-testid="inputConfirmPassword"
                className="input"
                placeholder="Confirm Password"
                value={confirmPassword}
                minLength="8"
                type="password"
                onChange={(e) => setConfirmPassword(e.target.value)}
                required
              />{" "}
              <br />
              <label className="label">Mobile Number</label>
              <br />
              <input
                data-testid="inputMobileNumber"
                className="input"
                placeholder="Mobile Number"
                value={mobileNumber}
                minLength="10"
                maxLength="10"
                type="text"
                onChange={(e) => setMobileNumber(e.target.value)}
                required
              />
              <br />
              <Button
                data-testid="submit-btn"
                className="btn"
                variant="primary"
                type="submit"
              >
                Submit
              </Button>
            </Form>
          </Col>
        </Row>
      </Container>
    </Layout>
  );
};

export default Signup;
