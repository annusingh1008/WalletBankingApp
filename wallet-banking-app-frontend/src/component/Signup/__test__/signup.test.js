import React from "react";
import { fireEvent } from "@testing-library/react";
import Signup from "../index";
import { localStorageMock, renderWithRedux } from "../../../helper/testHelper";
import { signup } from "../../../actions/user.actions";
import { Form } from "react-bootstrap";

const mockDispatch = jest.fn();
jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useDispatch: () => mockDispatch,
}));

jest.mock("../../Layout", () =>
  jest.fn(({ children }) => <div data-testid="Layout">{children}</div>)
);

// jest.mock("react-bootstrap", () => {
//   const Container = jest.fn(({ children }) => (
//     <table data-testid="Container">{children}</table>
//   ));
//   const Row = jest.fn(({ children }) => (
//     <div data-testid="Row">{children}</div>
//   ));
//   const Col = jest.fn(({ children }) => (
//     <div data-testid="Col">{children}</div>
//   ));
//   const Form = jest.fn(({ children }) => (
//     <div data-tesid="Form">{children}</div>
//   ));
//   const Button = jest.fn((props) => (
//     <button onClick={props.onClick} data-testid={`${props["data-testid"]}`}>
//       {props.children}
//     </button>
//   ));

//   return { Container, Row, Col, Form, Button };
// });

jest.mock("../../../actions/user.actions", () => ({
  signup: jest.fn(() => ({ type: "signup" })),
  getUser: jest.fn(() => ({
    error: null,
    message: "",
    loading: false,
  })),
}));

describe("Signup", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  Object.defineProperty(window, "localStorage", { value: localStorageMock });

  it("renders properly", () => {
    const { container } = renderWithRedux(<Signup />);
    expect(container.firstChild).toMatchSnapshot();
  });

  describe("When clicks on first button", () => {
    it("should render signup textarea element", () => {
      const { queryByTestId } = renderWithRedux(<Signup />);

      const textAreaElement1 = queryByTestId("inputFirstName");
      expect(textAreaElement1).toBeInTheDocument();

      const textAreaElement2 = queryByTestId("inputLastName");
      expect(textAreaElement2).toBeInTheDocument();

      const textAreaElement3 = queryByTestId("inputEmail");
      expect(textAreaElement3).toBeInTheDocument();

      const textAreaElement4 = queryByTestId("inputPassword");
      expect(textAreaElement4).toBeInTheDocument();

      const textAreaElement5 = queryByTestId("inputConfirmPassword");
      expect(textAreaElement5).toBeInTheDocument();

      const textAreaElement6 = queryByTestId("inputMobileNumber");
      expect(textAreaElement6).toBeInTheDocument();
    });

    it("should be able to type signup details", () => {
      const { queryByTestId } = renderWithRedux(<Signup />);

      const textAreaElement1 = queryByTestId("inputFirstName");
      fireEvent.change(textAreaElement1, {
        target: { value: "Annu" },
      });
      expect(textAreaElement1.value).toBe("Annu");

      const textAreaElement2 = queryByTestId("inputLastName");
      fireEvent.change(textAreaElement2, { target: { value: "Singh" } });
      expect(textAreaElement2.value).toBe("Singh");

      const textAreaElement3 = queryByTestId("inputEmail");
      fireEvent.change(textAreaElement3, {
        target: { value: "annu@gmail.com" },
      });
      expect(textAreaElement3.value).toBe("annu@gmail.com");

      const textAreaElement4 = queryByTestId("inputPassword");
      fireEvent.change(textAreaElement4, { target: { value: "1234" } });
      expect(textAreaElement4.value).toBe("1234");

      const textAreaElement5 = queryByTestId("inputConfirmPassword");
      fireEvent.change(textAreaElement5, { target: { value: "1234" } });
      expect(textAreaElement5.value).toBe("1234");

      const textAreaElement6 = queryByTestId("inputMobileNumber");
      fireEvent.change(textAreaElement6, { target: { value: "1234567890" } });
      expect(textAreaElement6.value).toBe("1234567890");
    });

    it("should have empty textarea when submit button is clicked", () => {
      const { queryByTestId } = renderWithRedux(<Signup />);

      const buttonElement = queryByTestId("submit-btn");

      const textAreaElement1 = queryByTestId("inputFirstName");
      fireEvent.change(textAreaElement1, {
        target: { value: "" },
      });

      const textAreaElement2 = queryByTestId("inputLastName");
      fireEvent.change(textAreaElement2, { target: { value: "" } });

      const textAreaElement3 = queryByTestId("inputEmail");
      fireEvent.change(textAreaElement3, {
        target: { value: "" },
      });

      const textAreaElement4 = queryByTestId("inputPassword");
      fireEvent.change(textAreaElement4, { target: { value: "" } });

      const textAreaElement5 = queryByTestId("inputConfirmPassword");
      fireEvent.change(textAreaElement5, { target: { value: "" } });

      const textAreaElement6 = queryByTestId("inputMobileNumber");
      fireEvent.change(textAreaElement6, { target: { value: "" } });

      fireEvent.click(buttonElement);

      expect(textAreaElement1.value).toBe("");
      expect(textAreaElement2.value).toBe("");
      expect(textAreaElement3.value).toBe("");
      expect(textAreaElement4.value).toBe("");
      expect(textAreaElement5.value).toBe("");
      expect(textAreaElement6.value).toBe("");
    });

    it("dispatches proper action", () => {
      const { container, queryByTestId } = renderWithRedux(<Signup />);
      const firstNameElement = queryByTestId("inputFirstName");
      const lastNameElement = queryByTestId("inputLastName");
      const emailElement = queryByTestId("inputEmail");
      const passwordElement = queryByTestId("inputPassword");
      const confirmPasswordElement = queryByTestId("inputConfirmPassword");
      const mobileNumberElement = queryByTestId("inputMobileNumber");
      fireEvent.change(firstNameElement, { target: { value: "Annu" } });
      fireEvent.change(lastNameElement, { target: { value: "Singh" } });
      fireEvent.change(emailElement, { target: { value: "annu@gmail.com" } });
      fireEvent.change(passwordElement, { target: { value: "1234" } });
      fireEvent.change(confirmPasswordElement, {
        target: { value: "abcdefgh" },
      });
      fireEvent.change(mobileNumberElement, {
        target: { value: "1234567890" },
      });
      expect(container.firstChild).toMatchSnapshot();
      const node = queryByTestId("submit-btn");
      fireEvent.click(node);
      fireEvent.submit(queryByTestId("submit-form"));
    });
  });
});
