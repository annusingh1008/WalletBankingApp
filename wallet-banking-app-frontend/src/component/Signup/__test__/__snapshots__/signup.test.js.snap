// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup When clicks on first button dispatches proper action 1`] = `
<div
  data-testid="Layout"
>
  <div
    class="container"
    style="margin-top: 110px;"
  >
    <div
      class="row"
      style="margin-left: 18%;"
    >
      <div
        class="col"
      >
        <form
          class=""
          data-testid="submit-form"
        >
          <label>
            First Name
          </label>
          <br />
          <input
            class="input"
            data-testid="inputFirstName"
            placeholder="First Name"
            required=""
            type="text"
            value="Annu"
          />
          <br />
          <label
            class="label"
          >
            Last Name
          </label>
          <br />
          <input
            class="input"
            data-testid="inputLastName"
            placeholder="Last Name"
            required=""
            type="text"
            value="Singh"
          />
           
          <br />
          <label
            class="label"
          >
            Email
          </label>
          <br />
          <input
            class="input"
            data-testid="inputEmail"
            placeholder="Email"
            required=""
            type="email"
            value="annu@gmail.com"
          />
           
          <br />
          <label
            class="label"
          >
            Password
          </label>
          <br />
          <input
            class="input"
            data-testid="inputPassword"
            minlength="8"
            placeholder="Password"
            required=""
            type="password"
            value="1234"
          />
           
          <br />
          <label
            class="label"
          >
            Confirm Password
          </label>
          <br />
          <input
            class="input"
            data-testid="inputConfirmPassword"
            minlength="8"
            placeholder="Confirm Password"
            required=""
            type="password"
            value="abcdefgh"
          />
           
          <br />
          <label
            class="label"
          >
            Mobile Number
          </label>
          <br />
          <input
            class="input"
            data-testid="inputMobileNumber"
            maxlength="10"
            minlength="10"
            placeholder="Mobile Number"
            required=""
            type="text"
            value="1234567890"
          />
          <br />
          <button
            class="btn btn btn-primary"
            data-testid="submit-btn"
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`Signup renders properly 1`] = `
<div
  data-testid="Layout"
>
  <div
    class="container"
    style="margin-top: 110px;"
  >
    <div
      class="row"
      style="margin-left: 18%;"
    >
      <div
        class="col"
      >
        <form
          class=""
          data-testid="submit-form"
        >
          <label>
            First Name
          </label>
          <br />
          <input
            class="input"
            data-testid="inputFirstName"
            placeholder="First Name"
            required=""
            type="text"
            value=""
          />
          <br />
          <label
            class="label"
          >
            Last Name
          </label>
          <br />
          <input
            class="input"
            data-testid="inputLastName"
            placeholder="Last Name"
            required=""
            type="text"
            value=""
          />
           
          <br />
          <label
            class="label"
          >
            Email
          </label>
          <br />
          <input
            class="input"
            data-testid="inputEmail"
            placeholder="Email"
            required=""
            type="email"
            value=""
          />
           
          <br />
          <label
            class="label"
          >
            Password
          </label>
          <br />
          <input
            class="input"
            data-testid="inputPassword"
            minlength="8"
            placeholder="Password"
            required=""
            type="password"
            value=""
          />
           
          <br />
          <label
            class="label"
          >
            Confirm Password
          </label>
          <br />
          <input
            class="input"
            data-testid="inputConfirmPassword"
            minlength="8"
            placeholder="Confirm Password"
            required=""
            type="password"
            value=""
          />
           
          <br />
          <label
            class="label"
          >
            Mobile Number
          </label>
          <br />
          <input
            class="input"
            data-testid="inputMobileNumber"
            maxlength="10"
            minlength="10"
            placeholder="Mobile Number"
            required=""
            type="text"
            value=""
          />
          <br />
          <button
            class="btn btn btn-primary"
            data-testid="submit-btn"
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  </div>
</div>
`;
